spark-scala/src/main/scala/com/matthewrathbone/spark/Main.scala

```diff
@@ -0,0 +1,35 @@
+package main.scala.com.matthewrathbone.spark
+
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapred.TextOutputFormat
+
+object Main {
+  def main(args: Array[String]) {
+	val conf = new SparkConf().setAppName("SparkJoins").setMaster("local")
+	val sc = new SparkContext(conf)
+	
+	val transactions = sc.textFile("transactions.txt")
+	val newTransactionsPair = transactions.map{t =>                
+	    val p = t.split("\t")
+	    (p(2).toInt, p(1).toInt)
+	}
+	
+	val users = sc.textFile("users.txt")
+	val newUsersPair = users.map{t =>                
+	    val p = t.split("\t")
+	    (p(0).toInt, p(3))
+	}
+	
+	val jn = newTransactionsPair.leftOuterJoin(newUsersPair).values.distinct
+	val result = jn.countByKey
```

Comment:
What do the data structures look like after the join?

After the join does it look something like `(key, (value1, value2))`?
Can we add a comment here to explain how these two lines work?